### ISSUE TITLE
missing FTP/SSH/NFS-support in 2.1

### ISSUE THREAD

**Issue by GralfR** [rescuezilla/rescuezilla#169]
strange thing: I've downloaded the 2.1.2 live-iso and bootet rescuezilla.
The GUI only offers CIFS-networkshare (which I did not get to connect by the way). Because it's a linux-system I expected a FTP or even NFS-connection, but the dropdown-menu did not offer any other than CIFS.
Inside the file-manager on the desktop there is a FTP/SSH/WebDAV, but no SMB/CIFS.
My server (a QNAP) has FTP, SSH, NFS and Samba-shares. All working great with other devices from Linux/WinXP/Win7/Win10 and other OSes for NFS.
Even though I was able to ping my server from the rescuezilla terminal (static IPv4 without gateway/direct route) I didn't manage to get a connection.
Finally I did a backup a local USB-drive. I can't say, what was wrong, but I apreachiate rescuezilla for the great idea of that package, with which I was able to resize a LVM2-partition with the gparted. Keep in mind: I'm a windows-guy and just started struggling with linux. Rescuezilla was very helpfull, but the network-connections should be improved. It's hard for a windows-guy to work on a linux-terminal to mount USB-drives or network-drives and if any package is missing it's even harder to install it without a internet-connection on that system. So, the easy way of rescuezilla is great, but missing easy network mounting.

Thanks

**Comment by shasheene** [comment]
Hi GralfR,

There are a number of existing issues capturing the tasks you have described:

#81: Add SSH File Transfer Protocol support (SFTP) 
#118: Add NFS support
#158: Backup and restoring from cloud providers (Dropbox, Google Drive etc) using rclone's FUSE mount

I hope to finally implement at least some of these features for Rescuezilla v2.2 scheduled for release in May 2021 (a few months away). Sorry it has taken so long, I have been focused on getting Image Explorer (beta) functionality out to users so additional network protocols have been lower priority. By the way, Rescuezilla previously had legacy FTP support but I removed it due to the issue described in #79 (in short, the "curlftpfs" client doesn't support appending to files which is a feature required for Rescuezilla to make backups).

**A few notes of SMB/CIFS support that might help you connect to your QNAP Samba share:**
* Some network shares may not support the default SMB 3.1.1 protocol, so you may need to enter '2.1' or even '1.0' in the version field.
* Make sure the fields have the correct UNC path (eg "\\fileserver\johnsmith").
    * I suggest just doing the "\\server\sharename" pattern and _not_ specifying the entire path even if you know it
* In earlier versions, some users had trouble with some Anonymous shares that didn't require a username/password but success with username/password

**Comment by GralfR** [comment]
Hi shasheene!

Thanks for the hints. Finally the version 1.0 did connect.
I wonder if this could be improoved inside the GUI. For a newbee and a windows guy it's not obvious, what to enter inside the optional fields. So beeing not sure I left them untouched/empty.
The error-message did not help that much. And often with a live-CD during backup you not always have access to the internet for forum research.

If I get it right, the version field only accepts "1.0", "2.1" and "3.1.1". Maybe then it's better not to have a text input but a radiobutton-selection, which is default nothing or the default version selected. With radiobuttons the user will get a guess, what this is for and just try the available options by click and eleminating syntax errors. 
_@admin: you might move this comment to open a new request, if this fits more the internal structure._

Thanks so far, my backup is running...

**Comment by shasheene** [comment]
Yep, you're right: the SMB/CIFS version field has a poor user experience because there no indication when the user needs to enter a custom version or the values which are accepted.

Maybe a radio button approach would work like you suggest. Or a dropdown menu. A custom text box will still be required because whilst right now there's only a number of versions that the field will accept, over time they might be an eg, SMB/CIFS v3.1.2 or v4.0 etc and keeping track of this is an overhead I'd like to avoid (Rescuezilla will support many more networking protocols soon).

Perhaps the easiest alternative is to display an error message after first connection that asks the user to try a set of version numbers. I'll experiment and try and improve things.

**Comment by chicks-net** [comment]
Should this ticket get renamed since the title is already covered in other tickets mentioned above?  And it seems like the focus of this ticket is now on the SMB/CIFS aspect.

**Comment by shasheene** [comment]
OK, I have made this into a dropdown menu for Rescuezilla v2.2 (to be released at the end of May 2021, ie in a few days):

![smb_version_dropdown](https://user-images.githubusercontent.com/1414379/119634219-ae402580-be51-11eb-8c56-bb7915e446bb.png)
